src/server/NetworkingServer.cpp

```diff
@@ -163,36 +180,53 @@ void NetworkingServer::ComputeGameState() {
         mAcceptingMoveCommands = true;
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
         mAcceptingMoveCommands = false;
-        // update game state, take moves array and simulate next tick given current state and inputs
-        // ... simplest behavior would be to track client locations and apply vectors to them ...
 
-        if (mClientMoves.size() == 0) {
-            continue;
-        }
-        std::cout << "game state thread: dumping client moves for tick " << mCurrTick << std::endl;
-        for (const auto iter : mClientMoves) {
-            sf::Uint32 client_id = iter.first;
-            float* move_vector = iter.second;
-            if (mClientPositions.find(client_id) != mClientPositions.end()) {
-                mClientPositions[client_id][0] += move_vector[0]*5.f; // x component
-                mClientPositions[client_id][1] += move_vector[1]*5.f; // y component
-            }
-            std::cout << iter.first << " " << iter.second[0] << " " << iter.second[1] << std::endl;
-        }
-        std::cout << "game state thread: dumping client positions for tick " << mCurrTick << std::endl;
-        for (const auto iter : mClientPositions) {
-            std::cout << iter.first << " " << iter.second[0] << " " << iter.second[1] << std::endl;
-        }
         // window is closed, time to reset. May want to use fix sized array as an optimization.
         // It's a bit mean to disguard old commands, maybe there should be windows for multiple ticks?
-        mClientMoves.clear();
         mCurrTick++;
+        // This sleep is needed to give the GameController time to fetch the client directions in getClientDirections
+        // TODO: Not sure what this sleep value should be...
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Not sure what this value should be. This is the window during which the server gets to read the move commands (mAcceptingMoveCommands == false). See getNetworkPlayers.

If I don't have a sleep here then mAcceptingMoveCommands is almost always true and the server doesn't get a chance to getNetworkPlayers and update player directions according to the client input.

Reply:
aha yeah nice catch. previously when the computation was happening in this function, it was a given that it would be false at the right time. Im curious why you pulled the computation of game state out of this code

Reply:
I don't want `NetworkServer` to call an update function in `GameController` from `ComputeGameState`. 
I don't want the `NetworkServer` to have a reference to `GameController` or even know of its existence.
I want the `NetworkServer` to be _used_ by the `GameController` to communicate state to and receive commands from the client.

Let's discuss this in https://github.com/SourenP/ungroup_sfml/pull/19/files/c5d2bc2e8609966792ecc862ac310ecdf5999c43#r289660647

Reply:
Moved the game state update to Gamecontroller in b5490ec